src/Analyzer.Utilities/Extensions/ISymbolExtensions.cs

```diff
@@ -147,6 +147,30 @@ public static bool MatchFieldByName(this ISymbol member, INamedTypeSymbol type,
         {
             return member != null && member.Kind == SymbolKind.Field && member.MatchMemberByName(type, name);
         }
+
```

Comment:
Move this method from the CA1716 analyzer into Analyzer.Utilities. All FxCop rules format the names of nested classes this way. Odds are there are other analyzers, already implemented, whose output wouldn't be exactly the same as FxCops when the fired against a nested class.